test/input-test.jl

```diff
@@ -0,0 +1,26 @@
+# Averaging the spike count of ConstantRate over a fixed window of time should be ConstantRate.rate
+@testset "ConstantRate" begin
+    T = 10_000
+    rate = rand()
+    input = ConstantRate(rate)
+    @test isapprox(count(x -> x > 0, [input(t) for t in 1:T]) / T, rate, atol=0.005)
+end
+
+# Assert frequency constructor matches the rate constructor
+@testset "ConstantRate Constructors" begin
+    freq, dt = rand(), rand()

```

Comment:
This won’t consistently work. Frequency is in Hertz. If `freq == 0.01`, then you need to enforce `dt <= 0.01`.

Reply:
I did add a condition but I am a bit confused about why we need this condition. Assuming that ` freq = 0.01` and `dt = 0.5` which makes `rate = 0.005` but given this condition the test does pass. So, I think I am missing something here and I am not sure if the way I dealt with this is correct.